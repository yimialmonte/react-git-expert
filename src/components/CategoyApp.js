import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoyApp = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 1) {
      setCategories((prev) => [...prev, inputValue]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInput} />
    </form>
  );
};

CategoyApp.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default CategoyApp;
