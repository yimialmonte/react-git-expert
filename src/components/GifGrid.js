import React, { useEffect, useState } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGif';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then(setImages);
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((item) => (
          <GifGridItem key={item.id} {...item} />
        ))}
      </div>
    </>
  );
};
