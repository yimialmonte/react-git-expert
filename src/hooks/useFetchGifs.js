import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGif';

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((resp) => {
      setTimeout(() => {
        setState({ data: resp, loading: false });
      }, 3000)
    });
  }, [category]);

  return state;
};
