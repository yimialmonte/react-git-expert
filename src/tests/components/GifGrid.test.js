import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Test for component GifGrid', () => {
  const category = 'Hand';

  test('Should show component correctly', () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show items when load imagenes', () => {
    const gifs = [{ id: 'ABC', url: 'https//localhost', title: 'Title' }];

    useFetchGifs.mockReturnValue({ data: gifs, loading: false });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

  });
});
