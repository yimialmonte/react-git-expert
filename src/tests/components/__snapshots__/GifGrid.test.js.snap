// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for component GifGrid Should show component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__lightSpeedInRight"
  >
    Hand
  </h3>
  <p
    className="animate__animated animate__bounce"
  >
     Cargando ...
  </p>
  <div
    className="card-grid animate__animated animate__fadeInDown"
  />
</Fragment>
`;

exports[`Test for component GifGrid Should show items when load imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__lightSpeedInRight"
  >
    Hand
  </h3>
  <div
    className="card-grid animate__animated animate__fadeInDown"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Title"
      url="https//localhost"
    />
  </div>
</Fragment>
`;
